### PR TITLE
Implement agenda topappbar  and dates

### PR DESCRIPTION
**Pull request by Gerondk**

### CODE REVIEW

File: agenda/presentation/src/main/java/com/gkp/agenda/presentation/AgendaUiState.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        if (selectedDate != other.selectedDate) return false
+
+        return true
+    }
+
+    override fun hashCode(): Int {
+        var result = fullName.hashCode()
+        result = 31 * result + selectedDate.hashCode()
+        return result
+    }
```
Comment (philipplackner) [inline agenda/presentation/src/main/java/com/gkp/agenda/presentation/AgendaUiState.kt]: Why are these needed here? Isn't `String` and `LocalDateTime` compared by value by default? I'd be careful with overriding equals and hashCode for states, since as soon as you add another field to this class and forget considering it in equals and hashCode, those functions will break. 99% of state classes should be able to use the default data class implementations of these.

File: agenda/presentation/src/main/java/com/gkp/agenda/presentation/compoments/DayItem.kt [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            .then(
+                if (isDateSelected) {
+                    Modifier.drawBehind {
+                        drawOval(
+                            color = TaskyOrange,
+                        )
+                    }
+                } else {
+                    Modifier
+                }
```
Comment (philipplackner) [inline agenda/presentation/src/main/java/com/gkp/agenda/presentation/compoments/DayItem.kt]: It's quite ugly having to use conditional modifiers with `then` (still the correct way). I personally like to create a custom modifier like this:
```
fun Modifier.applyIf(
    condition: Boolean,
    modifier: Modifier.() -> Modifier
): Modifier {
    return this.then(
        if (condition) {
            modifier()
        } else Modifier
    )
}
```
You can then do
```
modifier = Modifier
    .applyIf(isDateSelected) {
        drawBehind { ... }
    }
```

File: agenda/presentation/src/main/java/com/gkp/agenda/presentation/util/Util.kt [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import java.time.LocalDate
+private const val CURRENT_AND_NEXT_SIX_DATES = 7
+
+fun getFullNameInitials(fullName: String): String {
+    val names = fullName.split(" ")
+    val initialsBuilder = StringBuilder()
+    if (names.size == 1) {
+        initialsBuilder.append(names[0].take(2).uppercase())
+        return initialsBuilder.toString()
+    }
```
Comment (philipplackner) [inline agenda/presentation/src/main/java/com/gkp/agenda/presentation/util/Util.kt:18]: Might want to check the length of the first name as well. If it's <= 1, this will break

File: agenda/presentation/src/main/java/com/gkp/agenda/presentation/util/Util.kt [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    if (names.size == 2) {
+        initialsBuilder.append("${names[0].first()}${names[1].first()}".uppercase())
+        return initialsBuilder.toString()
+    }
+    if (names.size >= 3) {
+        initialsBuilder.append("${names[0].first()}${names[2].first()}".uppercase())
+        return initialsBuilder.toString()
+    }
```
Comment (philipplackner) [inline agenda/presentation/src/main/java/com/gkp/agenda/presentation/util/Util.kt]: Even though rare, a name with two middle names would not be considered, since `Philipp Middle1 Middle2 Lackner` would result in the initials `PM` here

File: agenda/presentation/src/main/java/com/gkp/agenda/presentation/util/Util.kt [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    return ""
+}
```
Comment (philipplackner) [inline agenda/presentation/src/main/java/com/gkp/agenda/presentation/util/Util.kt]: This function in general deals well with the happy path. In reality, the happy path is not always met however. A few considerations:
1. What about a name like `Philipp  Lackner` (with two spaces in between)? This would crash here, since it would result in a splitted list `["Philipp", "", "Lackner"]` where you then might try to access the first character of an empty string.
2. What about weird/invalid inputs? What if a user makes their name full of emojis? That's not explicitly in the requirements, but generally those would be valid inputs, so we as devs should also think about a nice and reasonable way to deal with that.

File: agenda/presentation/src/main/java/com/gkp/agenda/presentation/util/Util.kt [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+                )
+            )
+        }
+    }
+}
+
+data class DateWithSelected(
+    val date: LocalDate,
+    val isSelected: Boolean,
+)
```
Comment (philipplackner) [inline agenda/presentation/src/main/java/com/gkp/agenda/presentation/util/Util.kt:52]: While your preference how you create a list is completely up to you, just to share mine:
```
val today = LocalDate.now()
return (0 until CURRENT_AND_NEXT_SIX_DATES).map {
    DateWithSelected(
        date = today.plusDays(it.toLong()),
        isSelected == date == selectedDate
    )
}
```
Has one less indentation 😄

**Review by philipplackner — APPROVED** [review]

File: agenda/presentation/src/main/java/com/gkp/agenda/presentation/AgendaUiState.kt (hunk 1 above)
Comment (Gerondk) [reply agenda/presentation/src/main/java/com/gkp/agenda/presentation/AgendaUiState.kt]: I removed equals and hashCode. Before the compiler was complaining

File: agenda/presentation/src/main/java/com/gkp/agenda/presentation/util/Util.kt (hunk 5 above)
Comment (Gerondk) [reply agenda/presentation/src/main/java/com/gkp/agenda/presentation/util/Util.kt]: To overcome "1". I filtered the list again empty  and blank string
